src/EmployerPR/SFA.DAS.EmployerPR/Application/Commands/PostPermissions/PostPermissionsCommandHandler.cs

```diff
@@ -1,12 +1,142 @@
 ﻿using MediatR;
+using RestEase;
+using SFA.DAS.EmployerPR.Application.Commands.PostNotifications;
+using SFA.DAS.EmployerPR.Application.Queries.GetPermissions;
+using SFA.DAS.EmployerPR.Common;
 using SFA.DAS.EmployerPR.Infrastructure;
+using SFA.DAS.SharedOuterApi.Models.ProviderRelationships;
+using System.Net;
 
 namespace SFA.DAS.EmployerPR.Application.Commands.PostPermissions;
 
-public class PostPermissionsCommandHandler(IProviderRelationshipsApiRestClient _providerRelationshipsApiRestClient) : IRequestHandler<PostPermissionsCommand, PostPermissionsCommandResult>
+public class PostPermissionsCommandHandler(IProviderRelationshipsApiRestClient _providerRelationshipsApiRestClient) : IRequestHandler<PostPermissionsCommand, Unit>
 {
-    public async Task<PostPermissionsCommandResult> Handle(PostPermissionsCommand command, CancellationToken cancellationToken)
+    private const int PERMITS_APPROVAL = 1;
+    private const int PERMITS_RECRUITNENT = 1;
+    private const int PERMITS_RECRUITNENT_WITH_REVIEW = 2;
+
+    public async Task<Unit> Handle(PostPermissionsCommand command, CancellationToken cancellationToken)
+    {
+        Response<GetPermissionsResponse> permissionsResponse = 
+            await _providerRelationshipsApiRestClient.GetPermissions(
+                command.Ukprn, 
+                command.AccountLegalEntityId, 
+                cancellationToken
+        );
+
+        List<Operation> existingOperations = [];
+
+        switch(permissionsResponse.ResponseMessage.StatusCode)
+        {
+            case HttpStatusCode.OK:
+                {
+                    existingOperations = permissionsResponse.GetContent().Operations;
+                }
+                break;
+            case HttpStatusCode.NotFound:
+                {
+                    existingOperations = [];
+                }
+                break;
+            default:
+                {
+                    throw new InvalidOperationException($"Invalid operation occurred trying to retrieve permissions for ukprn {command.Ukprn} and AccountLegalEntityId {command.AccountLegalEntityId}");
```

Comment:
`Unexpected response code {permissionsResponse.ResponseMessage.StatusCode} returned when trying to ...`